CodeEdit/Features/Settings/Models/SettingsPage.swift

```diff
@@ -8,31 +8,31 @@
 import Foundation
 import SwiftUI
 
-/// A struct for a preferences tab
-struct SettingsPage: Hashable, Identifiable {
-    /// Default intializer
+/// A struct for a settings page
+struct SettingsPage: Hashable, Equatable, Identifiable {
+    /// Default initializer
     internal init(
         _ name: Name,
-        baseColor: Color? = nil,
-        icon: IconResource? = nil,
-        hideName: Bool? = false,
-        children: [SettingsPage] = []
+        baseColor: Color,
+        icon: IconResource,
+        isSetting: Bool = false,
```

Comment:
I don't like using this struct for both search results and settings pages, it'll get confusing. I think remove the `isSetting` property here and don't use this for search results.

Reply:
Unfortunately, that's not possible to change as the search results will group again, I have already tried to do so and it has taken an absurd amount of time. Sorry :(